wgpu-hal/src/gles/queue.rs

```diff
@@ -373,6 +373,64 @@ impl super::Queue {
             C::ClearStencil(value) => {
                 gl.clear_buffer_depth_stencil(glow::STENCIL, 0, 0.0, value as i32);
             }
+            C::BufferBarrier(raw, usage) => {
+                let mut flags = 0;
+                if usage.contains(crate::BufferUse::VERTEX) {
```

Comment:
From what I can tell from the gl spec, the buffers don't actually need to be involved when writing a barrier. The flags should all be concatenated and then one big gl.memory_barrier issued covering all buffers.

Reply:
Unfortunately, it's not that simple. The spec has wording explicitly saying which buffers are affected, e.g.:
>  The buffer objects affected by this bit are derived from the GL_ELEMENT_ARRAY_BUFFER binding.

Reply:
They sure hid that bit pretty well. Bleh, it pains me that this is how the spec is when most implementations only really have various kinds of global memory barriers. Ah well.